command/agent/template/template.go

```diff
@@ -255,8 +238,21 @@ func newRunnerConfig(sc *ServerConfig, templates ctconfig.TemplateConfigs) (*ctc
 		ServerName: pointerutil.StringPtr(""),
 	}
 
+	// The cache does its own retry management based on sc.AgentConfig.Retry,
+	// so we only want to set this up for templating if we're not routing
+	// templating through the cache.  We do need to assign something to Retry
+	// though or it will use its default of 12 retries.
+	var attempts int
+	if sc.AgentConfig.Vault != nil && sc.AgentConfig.Vault.Retry != nil {
+		attempts = sc.AgentConfig.Vault.Retry.NumRetries
+	}
+
 	// Use the cache if available or fallback to the Vault server values.
+	// For now we're only routing templating through the cache when persistence
+	// is enabled. The templating engine and the cache have some inconsistencies
+	// that need to be fixed for 1.7x/1.8
```

Comment:
Can you expand on the specific inconsitencies (backoff behavior, and perhaps turn the comment into a `// TODO: `?

Reply:
I don't know what they are, I'm just citing a slack convo with Jason.  I just put this comment in here because I was surprised that persistence was related to template use of cache, and decided to put his reply in here for the sake of others who might be similarly confused.